### PR TITLE
Amend the README to add more details about KMS key.

### PR DESCRIPTION
**Pull request by SteveLinden**
Added  links to relevant pull requests with code change details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,8 +73,22 @@ Under your team directory:
 - In *__data.tf__*, edit the data block changing the *__key_id__* to match the new key to be used in the pipeline.\
 *__default key:__* `key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-encryption-key"`\
 *__team key:__* `key_id = "arn:aws:kms:eu-west-2:${local.environment_management.account_ids["<team environemnt>"]}:alias/<team kms key alias name>"`\
+
 *__example of team key:__* `key_id = "arn:aws:kms:eu-west-2:${local.environment_management.account_ids["sprinkler-development"]}:alias/sprinkler_ebs-encryption-key"`
 
+More details of the required change can be seen in https://github.com/ministryofjustice/modernisation-platform-ami-builds/pull/18/files/6a589a6d3d0dc70f2bc28cb8cbb84075cad9d73c - ignore the rhel7 sections. \
```
Comment (davidkelliott) [inline README.md:79]: I think there is too much information in this PR for it to be useful as an example here

File: README.md [hunk 2]
```diff
@@ -73,8 +73,22 @@ Under your team directory:
...
+
+Should you have a shared kms key the above would not be required. If a default key is to be used this is already in place in the template, as below\
```
Comment (davidkelliott) [inline README.md:81]: There is a trailing \ at the end of a few lines

File: README.md [hunk 3]
```diff
@@ -73,8 +73,22 @@ Under your team directory:
...
+
+'data "aws_kms_key" "ebs_encryption_cmk" {key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-encryption-key"}'\
+
+If, however, a customer key is needed then the following needs to be set up
```
Comment (davidkelliott) [inline README.md:85]: This is covered in the section above "### How to add your own team kms key"

File: README.md [hunk 4]
```diff
@@ -73,8 +73,22 @@ Under your team directory:
...
+
+data "aws_kms_key" "sprinkler_ebs_encryption_key" {key_id = "arn:aws:kms:eu-west-2:${local.environment_management.account_ids["<ENV-NAME>"]}:alias/<KEY-NAME>"}
+
+*__sprinkler example__*
```
Comment (davidkelliott) [inline README.md:89]: This is covered already in the above section

File: README.md [hunk 5]
```diff
@@ -73,8 +73,22 @@ Under your team directory:
...
 *__default key:__* `key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-encryption-key"`\
 *__team key:__* `key_id = "arn:aws:kms:eu-west-2:${local.environment_management.account_ids["<team environemnt>"]}:alias/<team kms key alias name>"`\
+
 *__example of team key:__* `key_id = "arn:aws:kms:eu-west-2:${local.environment_management.account_ids["sprinkler-development"]}:alias/sprinkler_ebs-encryption-key"`
 
+More details of the required change can be seen in https://github.com/ministryofjustice/modernisation-platform-ami-builds/pull/18/files/6a589a6d3d0dc70f2bc28cb8cbb84075cad9d73c - ignore the rhel7 sections. \
+
+Should you have a shared kms key the above would not be required. If a default key is to be used this is already in place in the template, as below\
+
+'data "aws_kms_key" "ebs_encryption_cmk" {key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-encryption-key"}'\
```
Comment (davidkelliott) [inline README.md:83]: This wouldn't give the per business unit key, you would need something like:

`data "aws_kms_key" "ebs_encryption_cmk" {key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-<business-unit>"}`

For example:

`data "aws_kms_key" "ebs_encryption_cmk" {key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-hmpps"}`

**Review by davidkelliott — CHANGES_REQUESTED** [review]
Could the new information be put in it's own section please?  Something like:
"### How to use the per business unit shared kms key"

With a bit of information explaining that anyone in the business unit for that key can access and use it, and that images encrypted with the key will be accessible to all accounts in the business unit?  Thanks

**Review by davidkelliott — CHANGES_REQUESTED** [review]
I've added in some suggestions to make the guide read a little clearer and sort some formatting out.  After these are accepted I can approve 👍

**Review by davidkelliott — CHANGES_REQUESTED** [review]
It would be good to include an example of how to create a team kms key before showing how to reference it.  Something like:

Example code on how to create a team KMS key, and the permissions needed can be found [here](https://github.com/ministryofjustice/modernisation-platform-environments/blob/b73bba2e9d708efbc0db4492582829f52f00cb60/terraform/environments/sprinkler/kms.tf).

File: README.md [hunk 6]
```diff
@@ -73,8 +73,35 @@ Under your team directory:
...
+} 
+
+At the end of locals.tf include
+  ami_share_accounts = [
+    "${local.environment_management.account_ids["sprinkler-development"]}"
+  ]
+
+The above can be seen in the pull request https://github.com/ministryofjustice/modernisation-platform-ami-builds/pull/18/files/6a589a6d3d0dc70f2bc28cb8cbb84075cad9d73c# but this includes far more detail than is required here.
+
+
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply README.md:96]: Complete

File: README.md [hunk 7]
```diff
@@ -73,8 +73,35 @@ Under your team directory:
...
+Should you have a shared kms key the above would not be required. If a default key is to be used this is already in place in the template, as below
+
+data "aws_kms_key" "ebs_encryption_cmk" {key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-<business-unit>"}
+
+For example:
+
+data "aws_kms_key" "ebs_encryption_cmk" {key_id = "arn:aws:kms:eu-west-2:${data.aws_caller_identity.current.account_id}:alias/ebs-hmpps"}'
+
+If, however, a customer key is needed then follow the instructions in the section "### How to add your own team kms key".
```
[the remark this answers is not in the document]
Comment (SteveLinden) [reply README.md:103]: Completed

[the remark this answers is not in the document]
Comment (davidkelliott) [reply README.md:103]: I can still see this line there? I think it's not needed

**Review by davidkelliott — CHANGES_REQUESTED** [review]
It would be good to include an example of how to create a team kms key before showing how to reference it. This could go at the top of the team key section, something like:

Example code on how to create a team KMS key, and the permissions needed can be found [here](https://github.com/ministryofjustice/modernisation-platform-environments/blob/b73bba2e9d708efbc0db4492582829f52f00cb60/terraform/environments/sprinkler/kms.tf).

**Review by davidkelliott — APPROVED** [review]
👍

**Pull request merged**